draft-ietf-tls-dnssec-chain-extension-08.xml

```diff
@@ -117,7 +117,9 @@
       <xref target="RFC7671" /> of a TLS
       server without performing
       additional DNS record lookups and incurring the
-      associated latency penalty. It also provides the
+      associated latency penalty, but inherent difficulties
+      with having multiple distinct trust hierarchies can limit
+      its applicability in some situations. It also provides the
```

Comment:
The Denial of Existence is still needed for this mentioned use case:

Green field applications that are designed to always employ this extension, could of course unconditionally mandate its use (whether or not to the exclusion of PKIX authentication).

Reply:
Sorry, that comment means "this text needs to be rewritten so that it's clear"

Reply:
The specific point is that requiring the **extension** is not the same as requiring **TLSA records**. Some servers might not have TLSA records or DNSSEC.  Indeed with DNSSEC adoption still comparatively low (~9 million out of ~400 million domains) servers without DNSSEC can still participate in the green fields applications, which increases their usable scope while adoption of DNSSEC ramps up (or stalls).
Thus denial of existence is actually useful to avoid limiting applications to just the DNSSEC-signed server domains.
The green fields case is an a-priori pin for __t__ &isin; (-&infin;, &infin;)

Reply:
Fair enough.  Ben, would you like some suggested text, or would you like to take a first crack at it?

Reply:
I would normally want to take a crack at it, but am rather behind in telechat prep for this week, so we'd likely be waiting until Friday for that to happen. So, feel free to propose something.

Reply:
I've taken a stab at it in: https://github.com/vdukhovni/dnssec-chain-extension/commit/57163848b523311fddc9572d9b4b9b685ffa25b5

Reply:
Pushed an update for a typo noticed by Paul:  https://github.com/vdukhovni/dnssec-chain-extension/commit/74e11c516048c2a10b7774cfab57d22aa7c9f9a9

Reply:
Address comments from Nico: https://github.com/vdukhovni/dnssec-chain-extension/commit/f57a151961bf894db1b52acfe1aaccf7d9d50c9b

Reply:
The idea I had in mind with this chunk was that "[this extension] lets you do DANE authentication, but it's not perfect and the limitations that make it not perfect are rather inherent to the architecture" (because there are two distinct (sets of) trust anchors that are not trusted to make statements about the others' PKI).  This is the Introduction, so details are not super-appropriate, but it does seem appropriate to set out the scope that this mechanism has some inherent limitations.

It kind of seems like a lot of the discussion in this thread may have been intended to go on a different line?